### PR TITLE
Removed hardcoded value for machine pools to allow users to increase …

### PR DESCRIPTION
**Pull request by thaneunsoo**
…node count in the pool. Changed config struct to allow for NodesAndRoles to accept quantity as a value. Adjusted all tests using RKEMachinePoolSetup helper function to adjust to the new structure.

### CODE REVIEW

File: tests/framework/extensions/machinepools/machinepools.go [hunk 1]
```diff
@@ -46,25 +46,35 @@ func NewRKEMachinePool(controlPlaneRole, etcdRole, workerRole bool, poolName str
...
 // `machineConfig` is the *unstructured.Unstructured created by CreateMachineConfig
 // `nodeRoles` would be in this format
 //   []map[string]bool{
 //   {
-// 	   "controlplane": true,
-// 	   "etcd":         false,
-//     "worker":       false,
+// 	   ControlPlane: true,
+// 	   Etcd:         false,
+//     Worker:       false,
```
Comment (igomez06) [inline tests/framework/extensions/machinepools/machinepools.go:63]: Can you fix the indentation please

File: tests/framework/extensions/machinepools/machinepools.go [hunk 2]
```diff
@@ -46,25 +46,35 @@ func NewRKEMachinePool(controlPlaneRole, etcdRole, workerRole bool, poolName str
...
+//	   Quantity:     1,
 //   },
 //   {
-// 	   "controlplane": false,
-// 	   "etcd":         true,
-// 	   "worker":       false,
+// 	   ControlPlane: false,
+// 	   Etcd:         true,
+//     Worker:       false,
```
Comment (igomez06) [inline tests/framework/extensions/machinepools/machinepools.go:69]: Same here. no big deal just a nit pick

File: tests/v2/validation/provisioning/rke2/custom_cluster_test.go [hunk 3]
```diff
@@ -144,13 +145,19 @@ func (c *CustomClusterProvisioningTestSuite) ProvisioningRKE2CustomCluster(exter
...
 
-func (c *CustomClusterProvisioningTestSuite) ProvisioningRKE2CustomClusterDynamicInput(externalNodeProvider ExternalNodeProvider, nodesAndRoles []map[string]bool) {
+func (c *CustomClusterProvisioningTestSuite) ProvisioningRKE2CustomClusterDynamicInput(externalNodeProvider ExternalNodeProvider, nodesAndRoles []machinepools.NodeRoles) {
 	rolesPerNode := []string{}
 
 	for _, nodes := range nodesAndRoles {
 		var finalRoleCommand string
-		for role := range nodes {
-			finalRoleCommand += fmt.Sprintf(" --%s", role)
+		if nodes.ControlPlane == true {
```
Comment (igomez06) [inline tests/v2/validation/provisioning/rke2/custom_cluster_test.go:153]: You don't have to do the '==' you can just do `if nodes.ControlPlane` less verbose that way

**Review by igomez06 — COMMENTED** [review]
Looks really good, just some nitpicks then I'll give it my approval

**Review by slickwarren — CHANGES_REQUESTED** [review]

File: tests/v2/validation/provisioning/rke2/provisioning_node_driver_test.go [hunk 4]
```diff
@@ -245,6 +252,6 @@ func (r *RKE2NodeDriverProvisioningTestSuite) TestProvisioningDynamicInput() {
 
 // In order for 'go test' to run this suite, we need to create
 // a normal test function and pass our suite to suite.Run
-func TestRKE2ProvisioningTestSuite(t *testing.T) {
+func TestProvisioningTestSuite(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (igomez06) [reply tests/v2/validation/provisioning/rke2/provisioning_node_driver_test.go:255]: I agree

**Review by igomez06 — APPROVED** [review]
Looks good Tim, great job

File: tests/v2/validation/provisioning/README.md [hunk 5]
```diff
@@ -137,7 +137,18 @@ provisioningInput is needed to the run the RKE2 tests, specifically kubernetesVe
```
Comment (sowmyav27) [inline tests/v2/validation/provisioning/README.md:137]: I think this statement in the readme file will have to be changed? `nodes are divided by "|"` ?

**Review by sowmyav27 — CHANGES_REQUESTED** [review]

File: tests/v2/validation/provisioning/README.md [hunk 6]
```diff
@@ -133,11 +133,22 @@ RKE1 specifically needs a node template config to run properly. These are the in
 ---
 
 ### Provisioning Input
-provisioningInput is needed to the run the RKE2 tests, specifically kubernetesVersion, cni, and providers. nodesAndRoles is only needed for the TestProvisioningDynamicInput test, nodes are divided by "|". nodeProviders is only needed for custom cluster test; the framework only supports custom clusters through aws/ec2 instances.
+provisioningInput is needed to the run the RKE2 tests, specifically kubernetesVersion, cni, and providers. nodesAndRoles is only needed for the TestProvisioningDynamicInput test, node pools are divided by ",". nodeProviders is only needed for custom cluster test; the framework only supports custom clusters through aws/ec2 instances.
```
Comment (igomez06) [inline tests/v2/validation/provisioning/README.md:136]: Can you actually mention that is a list of objects separated by a comma. There is a slight mistake, but can you update it where it says `nodeProviders is only needed for custom cluster test` and change it to `nodeProviders is only needed for node provider cluster tests`

**Comment by igomez06** [comment]
Tim why are all the go.mod files updated?

**Review by igomez06 — DISMISSED** [review]

**Review by markusewalker — APPROVED** [review]
LGTM.

File: tests/v2/validation/provisioning/README.md (hunk 5 above)
Comment (sowmyav27) [reply tests/v2/validation/provisioning/README.md:137]: After an offline discussion - Can we change it to something like `“{nodeRoles : True/false, Quantity: <qty>},”`

**Review by sowmyav27 — APPROVED** [review]

**Pull request merged**